.changeset/spotty-toes-help.md

```diff
@@ -0,0 +1,17 @@
+---
+'@graphql-hive/yoga': patch
+---
+
+Align with GraphQL Yoga >=5.10.4 which has the relevant features mentioned below;
+
+- Remove \`tiny-lru\` dependency, and use `createLruCache` from `graphql-yoga`
+- - Yoga already provides a LRU cache implementation, so we can use that instead of having a separate dependency
```

Comment:
what is the difference between Yoga's implementation and `tiny-lru`?

Reply:
Yoga uses `lru-cache` package, and I don't think there is a big difference. My idea is to avoid introducing an extra dependency in the plugin but use the one Yoga provides to reduce the duplicate code.

Reply:
if there is no difference, why does Yoga adds this to itself?

Reply:
In Yoga, we use LRU caching for parsing and validation caching, and we preferred `lru-cache` at some point. 
But this plugin introduces a new dependency `tiny-lru` which does the same job. In order to avoid introducing new dependencies doing the same job, and it is an unnecessary dependency especially for the bundle size. But instead, Yoga exports `createLRUCache` function for plugins to re-use the same thing Yoga uses to avoid extra dependencies, code etc.

Reply:
yes I understand
I'm just asking, if the fact that now you expose this functionality means we need to maintain it and people will rely on it, and if so, is it really yoga's job?

Reply:
how important is the bundle size here?

Reply:
> how important is the bundle size here?

In Yoga, we always avoided to introduce new dependencies, duplicate code etc for bundle size. Extra dependencies mean extra bundle size not just for Yoga but also for Gateway which has this package built-in.

> I'm just asking, if the fact that now you expose this functionality means we need to maintain it and people will rely on it, and if so, is it really yoga's job?

We already maintain LRU caching in Yoga. So I don't think it is too much work here. It is already there in Yoga and used by many plugins.

Reply:
For me any extra non-optimal thing is important because when they become together, it becomes an issue that we need to solve later by analyzing each of these. I just want to prevent them growing before.